tools/building.py

```diff
@@ -606,6 +606,8 @@ def closure_compiler(filename, advanced=True, extra_closure_args=None):
   args += ['--language_out', 'NO_TRANSPILE']
   # Tell closure never to inject the 'use strict' directive.
   args += ['--emit_use_strict=false']
+  # Always output UTF-8 files, this helps generate UTF-8 code points instead of escaping code points with \uxxxx inside strings. https://github.com/google/closure-compiler/issues/4158
+  args += ['--charset=UTF8']
```

Comment:
I'd like to see this land separately too if possible, along with a test that closure no longer produces `\uxxx` for UTF-8 strings in the input.

Reply:
This effect is already verified by the existing tests at https://github.com/emscripten-core/emscripten/pull/21478/files#diff-95c5e9a22c0a073731d32eb5c839692fd5851db19b919d30f14c2fde0c7c261bR2-R5, so we should be covered?

Reply:
I'd still love to see this change (along with the test it effects) land separately, but I won't force you to do it.

Reply:
Oh wait the change in file size you are referring to is due to `SINGLE_FILE_BINARY_ENCODE`, right?

I was wondering if this change would effect other tests.. In theory it should effect any JS code which includes a UTF-8 string, right?  Regardless of `SINGLE_FILE` usage

Reply:
> I was wondering if this change would effect other tests.. In theory it should effect any JS code which includes a UTF-8 string, right? Regardless of `SINGLE_FILE` usage

Yeah, this would specify the encoding to be UTF-8 for all output when pushed through Closure.